### DISCUSSION TITLE
Coqui STT WebAssembly?

### DISCUSSION THREAD

**Discussion by dnaaun** [coqui-ai/STT#1996] [Q&A] [answered]
Thanks for this great package!  It looks much more ready to integrate into a production level environment when compared to alternative STT repos on Github.

My question is, has there been any effort, or conversation towards, compiling this repo for a WebAssembly target?

**Comment by reuben** [comment, marked as answer]
A long time ago someone tried it with an asm.js backend, but the ecosystem was way less mature back then. I don't think anyone has made a recent attempt.

**Comment by dnaaun** [comment]
Ah I see. I guess I will have to look into doing it myself. This is a very
general/vague request, but if you have any tips as to how to get started,
etc(I just discovered this repo), it would be much appreciated.

**Reply by reuben** [reply]
The first thing to figure out would be how to build for WebAssembly with Bazel, in particular how's support for that in the TensorFlow/TFLite build system. Then you could try building libstt.so for WASM and go from there.

**Reply by dnaaun** [reply]
Thanks for that. I'm currently looking into compiling speechbrain's models with TorchScript, that also seems to be no easy feat. I'll let you know if I make any progress on getting Coqui's models working in WASM.

**Comment by lissyx** [comment]
Back in DeepSpeech time, I bugged @bnjbvr quite a lot on that topic, but there was no threading and SIMD support.

**Reply by bnjbvr** [reply]
Fwiw, wasm threading has since shipped to Chrome/Firefox by default, and I _think_ SIMD too (at least enabled in Firefox standard on my machine, and I think Chrome's been using it for a long time to blur the backgrounds on google meet).
